doc/Language/grammar_tutorial.pod6

```diff
@@ -567,6 +567,40 @@ Here's the final code:
     }
     =end code
 
+=head2 Add actions directly
+
+Above we see how to associate grammars with action objects and perform
+actions on the match object. However, when we want to deal with the match
+object, that isn't the only way. See the example below:
+
+    =begin code
+    grammar G {
+      rule TOP { <function-define> }
+      rule function-define {
+        'sub' <identifier> {
```

Comment:
I really don't like the indentation of the code block here, as it suggests a relation to the previous regex atoms that isn't there.

Most regexes/grammars I've seen or written put the curly braces of multi-line blocks on separate lines, which makes them stand out as they should.

For single-line code blocks, it's ok to put the curlies on the same line as the code block itself:

```
    { say "end " ~ $/.made; }
````

Reply:
I try to rewrite it following the style at https://github.com/moritz/perl-6-regex-code/blob/master/chapter-09-grammars/08-symbol-table.p6

BTW, looking forward to your new book!